### PR TITLE
Add support for attachments

### PR DESCRIPTION
**Pull request by tetsuo**

### CODE REVIEW

File: src/couchify.ts [hunk 1]
```diff
@@ -81,132 +94,151 @@ export function couchify(options: CouchifyOptions) {
...
-    } = options
-
+    const attachmentsDir = path.join(baseDocumentsDir, options.attachmentsDir)
     const whitelistedRootDirs = [filtersDir, listsDir, showsDir, updatesDir, viewsDir]
-
     const envExtensions = Object.keys(require.extensions).map(d => d.substring(1))
+    const globOptions = { ignore: options.globIgnorePatterns, nodir: true }
+    const globMatchPattern = `+(${whitelistedRootDirs.join('|')})/*.{${envExtensions}}`
 
     return new Promise((resolve, reject) => {
```
Comment (Avaq) [inline src/couchify.ts]: You can remove this extra layer of nesting, remove all code like `.catch(reject)` and replace `resolve(x)` by `return x`.

Comment (tetsuo) [reply src/couchify.ts]: Done.

File: src/couchify.ts [hunk 2]
```diff
@@ -118,9 +121,28 @@ export function couchify(options: CouchifyOptions) {
 
                         return entry
                     })
+                        .then(res => res)
```
Comment (Avaq) [inline src/couchify.ts]: We can remove this line

File: src/couchify.ts [hunk 3]
```diff
@@ -118,9 +121,28 @@ export function couchify(options: CouchifyOptions) {
...
+                        .catch((designTaskErr: Error) => {
+                            console.warn('could not extract design function: ' + designTaskErr.message)
+                            throw designTaskErr
```
Comment (Avaq) [inline src/couchify.ts]: How is this error going to be handled? Depending on the answer we might be able to remove the `console` line above.

Comment (tetsuo) [reply src/couchify.ts]: Ok, I am bubbling it all the way up to `cmd.js` now, which looks something like this:

```
λ dev/couchify node cmd.js --name test --db example fixtures/attachments --dry                                 
could not couchify: /Users/onur/dev/couchify/fixtures/attachments/shows/simple.js: Unexpected token, expected ; (1:6) while parsing file: /Users/onur/dev/couchify/fixtures/attachments/shows/simple.js
```

**Pull request merged**